client/config_update.go

```diff
@@ -7,15 +7,21 @@ import (
 	"github.com/moby/moby/api/types/swarm"
 )
 
+// ConfigUpdateOptions holds options for updating a config.
+type ConfigUpdateOptions struct {
+	Version swarm.Version
+	Config  swarm.ConfigSpec
+}
+
 // ConfigUpdate attempts to update a config
-func (cli *Client) ConfigUpdate(ctx context.Context, id string, version swarm.Version, config swarm.ConfigSpec) error {
+func (cli *Client) ConfigUpdate(ctx context.Context, id string, options ConfigUpdateOptions) error {
```

Comment:
for this one, I wonder if version should stay a positional arg, because it's really required. 

Interested to hear what others think though /cc @corhere